src/core/typing.ml

```diff
@@ -3,49 +3,47 @@
 open Extra
 open Console
 open Terms
-open Print
 open Unif
 
-(** [check builtins ctx t a] tells whether [t] has type [a] in context
+(** [check ctx t a] tells whether [t] has type [a] in context
    [ctx]. *)
-let check : sym StrMap.t -> ctxt -> term -> term -> bool =
-  fun builtins ctx t a ->
+let check : ctxt -> term -> term -> bool = fun ctx t a ->
   let to_solve = Infer.check ctx t a in
-  match solve builtins {no_problems with to_solve} with
+  match solve {empty_problem with to_solve} with
   | None     -> false
   | Some([]) -> true
   | Some(cs) ->
+      let pp_constr = Print.pp_constr in
       List.iter (fatal_msg "Cannot solve %a.\n" pp_constr) cs; false
 
-(** [infer_constr builtins ctx t] tries to infer a type [a], together with
+(** [infer_constr ctx t] tries to infer a type [a], together with
    unification constraints [cs], for the term [t] in context [ctx].  The
    function returns [Some(a,cs)] in case of success, and [None] otherwise. *)
-let infer_constr
-    : sym StrMap.t -> ctxt -> term -> (term * unif_constrs) option =
-  fun builtins ctx t ->
+let infer_constr : ctxt -> term -> (term * constr list) option =
+  fun ctx t ->
   let (a, to_solve) = Infer.infer ctx t in
-  Option.map (fun cs -> (a, cs))
-    (solve builtins {no_problems with to_solve})
+  Option.map (fun cs -> (a, cs)) (solve {empty_problem with to_solve})
 
-(** [infer builtins ctx t] tries to infer a type [a] for [t] in the context
+(** [infer ctx t] tries to infer a type [a] for [t] in the context
    [ctx]. The function returns [Some(a)] in case of success, and [None]
    otherwise. *)
-let infer : sym StrMap.t -> ctxt -> term -> term option =
-  fun builtins ctx t ->
-  match infer_constr builtins ctx t with
+let infer : ctxt -> term -> term option = fun ctx t ->
+  match infer_constr ctx t with
   | None       -> None
   | Some(a,[]) -> Some(a)
   | Some(_,cs) ->
+      let pp_constr = Print.pp_constr in
       List.iter (fatal_msg "Cannot solve %a.\n" pp_constr) cs; None
 
-(** [sort_type builtins ctx t] checks that the type of the term [t] in context
+(** [sort_type ctx t] checks that the type of the term [t] in context
    [ctx] is a sort. If that is not the case, the exception [Fatal] is
    raised. *)
-let sort_type : sym StrMap.t -> ctxt -> term -> unit =
-  fun builtins ctx t ->
-  match infer builtins ctx t with
-  | None    -> fatal None "Unable to infer a sort for %a." pp t
+let sort_type : ctxt -> term -> unit = fun ctx t ->
+  let pp_term = Print.pp_term in
+  match infer ctx t with
+  | None    -> fatal None "Unable to infer a sort for %a." pp_term t
   | Some(a) ->
   match unfold a with
   | Type | Kind -> ()
-  | a           -> fatal None "%a has type %a (not a sort)." pp t pp a
+  | a           ->
+      fatal None "%a has type %a (not a sort)." pp_term t pp_term a
```

Comment:
And we see here that the short `pp` was useful to save us some lines. I think it makes sense to have it because it is such a common function (and there can be no error due to type-checking).